### ISSUE TITLE
IDBKeyRange queries don't work with multi-field indexes

### ISSUE THREAD

**Issue by robertfall** [treojs/treo-websql#2]
While trying to figure out why our queries were failing in Safari I noticed that [this function](https://github.com/treojs/treo-websql/blob/master/lib/index.js#L88-L93) does not cater for multi-field indexes when converting an IDBKeyRange for use with WebSQL.

```
function testValue(v) {
  return (((!r.lowerOpen && v >= r.lower) || (r.lowerOpen && v > r.lower)) && ((!r.upperOpen && v <= r.upper) || (r.upperOpen && v < r.upper))
    || (r.upper === undefined && ((!r.lowerOpen && v >= r.lower) || (r.lowerOpen && v > r.lower)))
    || (r.lower === undefined && ((!r.upperOpen && v <= r.upper) || (r.upperOpen && v < r.upper))));
}`
```

I suspect that using [this function](https://github.com/treojs/treo-websql/blob/master/lib/shim.js#L637-L688) to translate IDBKeyRanges as mentioned in #1 will actually fix this.

If #1 is not a simple fix or won't work for these issues I'm happy to submit a PR to allow `testValue` to support multi-field IDBKeyRanges.

**Comment by alekseykulikov** [comment]
Hi @robertfall! Let's try your solution. I would be also happy to merge your PR.
Just need to make sure, that treo's test-suite passes with this update.

**Comment by alekseykulikov** [comment]
closed by #3
